lib/node_modules/@stdlib/array/base/mskfilter-map/docs/types/index.d.ts

```diff
@@ -0,0 +1,171 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Array2D, Collection } from '@stdlib/types/array';
+
+/**
+* Array element indices.
+*/
+type Indices = [ number, number ];
+
+/**
+* Callback invoked for each array element.
+*
+* @returns result
+*/
+type Nullary<U, V> = ( this: V ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - array element
+* @returns result
+*/
+type Unary<T, U, V> = ( this: V, value: T ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - array element
+* @param indices - current array element indices
+* @returns result
+*/
+type Binary<T, U, V> = ( this: V, value: T, indices: Indices ) => U;
```

Comment:
I believe this should only be `index`, as we are only providing one index, that of the input array.

```suggestion
type Binary<T, U, V> = ( this: V, value: T, index: number ) => U;
```

This applies here and below. And also means we should get rid of the `Indices` type def above.